Source/CarthageKitTests/VersionSpec.swift

```diff
@@ -64,83 +55,100 @@ class VersionSpecifierSpec: QuickSpec {
-				expect(specifier.isSatisfied(by: versionTwoOne.pinnedVersion!)) == true
-				expect(specifier.isSatisfied(by: versionTwoTwo.pinnedVersion!)) == true
-				expect(specifier.isSatisfied(by: versionThree.pinnedVersion!)) == true
+				let specifier = VersionSpecifier.atLeast(SemanticVersion.fromPinnedVersion(v2_1_1).value!)
+				expect(specifier.isSatisfied(by: v1_3_2)) == false
+				expect(specifier.isSatisfied(by: v2_0_2)) == false
+				expect(specifier.isSatisfied(by: v2_1_1)) == true
+				expect(specifier.isSatisfied(by: v2_2_0)) == true
+				expect(specifier.isSatisfied(by: v3_0_0)) == true
 			}
 
 			it("should allow greater or equal minor and patch versions for .compatibleWith") {
-				let specifier = VersionSpecifier.compatibleWith(versionTwoOne)
-				expect(specifier.isSatisfied(by: versionOne.pinnedVersion!)) == false
-				expect(specifier.isSatisfied(by: versionTwoZero.pinnedVersion!)) == false
-				expect(specifier.isSatisfied(by: versionTwoOne.pinnedVersion!)) == true
-				expect(specifier.isSatisfied(by: versionTwoTwo.pinnedVersion!)) == true
-				expect(specifier.isSatisfied(by: versionThree.pinnedVersion!)) == false
+				let specifier = VersionSpecifier.compatibleWith(SemanticVersion.fromPinnedVersion(v2_1_1).value!)
+				expect(specifier.isSatisfied(by: v1_3_2)) == false
+				expect(specifier.isSatisfied(by: v2_0_2)) == false
+				expect(specifier.isSatisfied(by: v2_1_1)) == true
+				expect(specifier.isSatisfied(by: v2_2_0)) == true
+				expect(specifier.isSatisfied(by: v3_0_0)) == false
 			}
 
 			it("should only allow exact versions for .exactly") {
-				let specifier = VersionSpecifier.exactly(versionTwoTwo)
-				expect(specifier.isSatisfied(by: versionOne.pinnedVersion!)) == false
-				expect(specifier.isSatisfied(by: versionTwoZero.pinnedVersion!)) == false
-				expect(specifier.isSatisfied(by: versionTwoOne.pinnedVersion!)) == false
-				expect(specifier.isSatisfied(by: versionTwoTwo.pinnedVersion!)) == true
-				expect(specifier.isSatisfied(by: versionThree.pinnedVersion!)) == false
+				let specifier = VersionSpecifier.exactly(SemanticVersion.fromPinnedVersion(v2_2_0).value!)
+				expect(specifier.isSatisfied(by: v1_3_2)) == false
+				expect(specifier.isSatisfied(by: v2_0_2)) == false
+				expect(specifier.isSatisfied(by: v2_1_1)) == false
+				expect(specifier.isSatisfied(by: v2_2_0)) == true
+				expect(specifier.isSatisfied(by: v3_0_0)) == false
 			}
 
 			it("should allow only greater patch versions to satisfy 0.x") {
-				let specifier = VersionSpecifier.compatibleWith(versionZeroOne)
-				expect(specifier.isSatisfied(by: versionZeroOneOne.pinnedVersion!)) == true
-				expect(specifier.isSatisfied(by: versionZeroTwo.pinnedVersion!)) == false
+				let specifier = VersionSpecifier.compatibleWith(SemanticVersion.fromPinnedVersion(v0_1_1).value!)
+				expect(specifier.isSatisfied(by: v0_1_0)) == false
+				expect(specifier.isSatisfied(by: v0_1_1)) == true
+				expect(specifier.isSatisfied(by: v0_2_0)) == false
```

Comment:
Looks like those do test the same patch version and lower patch versions but does not test `greater patch versions` now.